test/E2ETests/SmokeTestsOnNanoServer.cs

```diff
@@ -0,0 +1,186 @@
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using E2ETests.Common;
+using Microsoft.AspNetCore.Server.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace E2ETests
+{
+    public class SmokeTestsOnNanoServer : IDisposable
+    {
+        private readonly XunitLogger _logger;
+        private readonly RemoteDeploymentConfig _remoteDeploymentConfig;
+
+        public SmokeTestsOnNanoServer(ITestOutputHelper output)
+        {
+            _logger = new XunitLogger(output, LogLevel.Information);
+
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("remoteDeploymentConfig.json")
+                .AddEnvironmentVariables()
+                .Build();
+
+            _remoteDeploymentConfig = new RemoteDeploymentConfig();
+            configuration.GetSection("NanoServer").Bind(_remoteDeploymentConfig);
+        }
+
+        [ConditionalTheory, Trait("E2Etests", "NanoServer")]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [SkipIfEnvironmentVariableNotEnabled("RUN_TESTS_ON_NANO")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64)]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64)]
+        public async Task Test(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture)
+        {
+            await RunTestsAsync(serverType, runtimeFlavor, architecture);
+        }
+
+        private async Task RunTestsAsync(
+            ServerType serverType,
+            RuntimeFlavor donetFlavor,
+            RuntimeArchitecture architecture)
+        {
+            using (_logger.BeginScope("SmokeTestSuite"))
+            {
+                var deploymentParameters = new RemoteDeploymentParameters(
+                    Helpers.GetApplicationPath(),
+                    serverType,
+                    donetFlavor,
+                    architecture,
+                    _remoteDeploymentConfig.FileSharePath,
+                    _remoteDeploymentConfig.ServerName,
+                    _remoteDeploymentConfig.AccountName,
+                    _remoteDeploymentConfig.AccountPassword,
+                    _remoteDeploymentConfig.ExecutableRelativePath)
+                {
+                    PublishTargetFramework = donetFlavor == RuntimeFlavor.Clr ? "net451" : "netstandardapp1.5",
+                };
+                deploymentParameters.EnvironmentVariables.Add(new KeyValuePair<string, string>("ASPNETCORE_ENVIRONMENT", "SocialTesting"));
+
+                using (var deployer = new RemoteDeployer(deploymentParameters, _logger))
+                {
+                    var deploymentResult = deployer.Deploy();
```

Comment:
This class https://github.com/aspnet/MusicStore/blob/dev/test/E2ETests/SmokeTests.cs seems to have a lot of duplicated code. Would it be possible to reuse the code by making the deployer a parameter?